homework-submission/barbarosapp/src/App.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from "react";
+import "./App.css";
+import logo from "./logo.svg";
+import ItemList from "./components/itemList";
+import Data from "./data/data";
+import AddMenu from "./components/addMenu";
+
+class App extends Component {
+  state = { Data };
+
+  handleCheck = id => {
+    const newData = this.state.Data.map(item => {
+      if (item.id === id) {
+        return { ...item, done: !item.done };
+      }
+      return item;
+    });
+
+    this.setState({ Data: newData });
+  };
+
+  handleRemove = id => {
+    function filterByID(item) {
+      if (item.id !== id) {
+        return true;
+      }
+      return false; 
+    }
+
+    const deletedData = this.state.Data.filter(filterByID);
+    this.setState({ Data: deletedData });
+  };
+
+  handleAdd = (name, deadline) => {
+    const newId = this.state.Data.length;
+    const newData = this.state.Data;
```

Comment:
Correction: this is not a new but it's the current state of data, just not making you confused with passing by `value` or `reference`.

Reply:
Yes, corrected